app/main/views/frameworks.py

```diff
@@ -330,3 +331,97 @@ def get_framework_interest(framework_slug):
     supplier_ids = [supplier_framework.supplier_id for supplier_framework in supplier_frameworks]
 
     return jsonify(interestedSuppliers=supplier_ids), 200
+
+
+@main.route('/frameworks/transition-dos/<string:framework_slug>', methods=['POST'])
+def transition_dos_framework(framework_slug):
+    updater_json = validate_and_return_updater_request()
+    json_payload = get_json_from_request()
+    json_has_required_keys(json_payload, ['expiringFramework'])
+
+    going_live_framework = Framework.query.filter(
+        Framework.slug == framework_slug
+    ).first_or_404()
+    expiring_framework = Framework.query.filter(
+        Framework.slug == json_payload['expiringFramework']
+    ).first_or_404()
+
+    if going_live_framework.id <= expiring_framework.id:
+        abort(400, f"'going_live_framework' ID ('{going_live_framework.id}') must greater than"
+              f"'expiring_framework' ID ('{expiring_framework.id}')")
+
+    if any(f.framework != 'digital-outcomes-and-specialists' for f in (going_live_framework, expiring_framework)):
+        abort(400, f"'going_live_framework' family: '{going_live_framework.framework}' and 'expiring_framework' family:"
+              f" '{expiring_framework.framework}' must both be 'digital-outcomes-and-specialists'")
+
+    if going_live_framework.status != 'standstill' or expiring_framework.status != 'live':
+        abort(400, f"'going_live_framework' status ({going_live_framework.status}) must be 'standstill', and "
+              f"'expiring_framework' status ({expiring_framework.status}) must be 'live'")
+
+    expiring_framework_draft_briefs = Brief.query.filter(
+        Brief.status == 'draft',
+        Brief.framework_id == expiring_framework.id,
+    ).all()
+
+    for brief in expiring_framework_draft_briefs:
+        brief.framework_id = going_live_framework.id
+        db.session.add(brief)
+        db.session.flush()
+
+        audit = AuditEvent(
+            audit_type=AuditTypes.update_brief_framework_id,
+            user=updater_json['updated_by'],
+            data={
+                'briefId': brief.id,
+                'previousFrameworkId': expiring_framework.id,
+                'newFrameworkId': going_live_framework.id,
+            },
+            db_object=brief,
+        )
```

Comment:
I would consider giving all these audit events (including those below) a common uniform timestamp.

Reply:
These should all occur within a very short time period. What would the benefit of a common timestamp be rather than when it actually happens?

Reply:
I suppose to kinda record that the logical order these happened in was basically simultaneous. Because it basically is, seeing that all the changes only become visible to the rest of the world at the same time.

Reply:
hmmm....interesting. That does make sense.

Reply:
After some jiggery-pokery I've realised that AuditEvents don't take a timestamp on it, they just use a default.
Rather than alter the model, I'm just going to leave as is. I don't think it's going to be an issue.

Reply:
Yeah I've run into that before the issue is just that the *constructor* doesn't expose the timestamp as an argument. You can still set the timestamp on the instance just after you've constructed it, which is what I've done elsewhere when I've done this.

Reply:
grumble grumble messy grumble